src/Orc.SystemInfo/Wmi/WindowsManagementConnection.cs

```diff
@@ -23,7 +22,7 @@ public void Open()
         {
             try
             {
-                ThrowIfDisposed();
+                CheckDisposed();
```

Comment:
is this method throwing an exception?
maybe it would be safer to return false instead of throwing an exception?

Reply:
Why? We sure we don't really want to allow this.
1) What's the benefit of hiding unappropriate behavior, returning boolean here? 
2) If we decide to return boolean here, how exactly you will differ behavior between 'Connection cannot be open, because it's disposed already, you are doing something wrong' and 'internal exception here, it's problem but not your fault'?
I also note here: **none** of simillar .NET apis (using-disposable-connection pattern) returns state in Open(), they are basically expose their state via Query/Transaction/Public property.